src/lib/markbind/src/parser.js

```diff
@@ -266,7 +285,19 @@ Parser.prototype._preprocess = function (node, context, config) {
     childContext.cwf = filePath;
     childContext.source = isIncludeSrcMd ? 'md' : 'html';
     if (element.children && element.children.length > 0) {
-      element.children = element.children.map(e => self._preprocess(e, childContext, config));
+      try {
+        element.children = element.children.map(e => self._preprocess(e, childContext, config));
+      } catch (e) {
+        if (e.message === 'Maximum call stack size exceeded') {
+          const err = new CyclicReferenceError();
+          err.addFileToStack(childContext.cwf);
+          throw err;
+        } else if (e instanceof CyclicReferenceError) {
+          e.addFileToStack(childContext.cwf);
+          throw e;
+        }
+        throw e;
```

Comment:
Interesting implementation. Let's move `CyclicReferenceError` into its own file and implement a `CyclicReferenceError.handle` class method, so that we can do this:

```js
try {
  element.children = element.children.map(e => self._preprocess(e, childContext, config));
} catch (e) {
  CyclicReferenceError.handle(e);
}
```

However, this implementation still waits for a call stack error to occur and depends on a hardcoded `e.message` check.
I wonder if a `FileManager` (suggested in #589) would help. We can proactively track and ensure acyclic file references.

Reply:
> I wonder if a `FileManager` (suggested in #589) would help.

I agree with letting FileManager handling this problem. @sijie123 might want to consider #589 first?

Reply:
Hmm, yea, I agree that it's kinda odd to wait for the error to occur and then checking, where we wcould have done something proactively first.

However, in that case FileManager would have to implement a subroutine similar to preprocess in order to detect any cyclic references, or even unused files for that matter.
I think it might be easier for me to implement a solution to the cyclic reference problem, then apply it to FileManager later.